### DISCUSSION TITLE
K12 Use Case

### DISCUSSION THREAD

**Discussion by johnfaig-stpats** [mozilla/hubs#3751] [General Questions & Discussion] [answered]
@johnshaughnessy, it was nice "meeting" you today.  Hubs is a powerful platform, but some of the power makes it difficult to use in a K12 environment.  This would be a game-changer because there are almost no EdTech tools that can be used by a wide range of grade levels.  Moving in the X, Y, and Z axes would be confusing for our younger students that don't play video games.  In this use case, a top-down view would look more like a classroom and still allow for different furniture configurations.   Further, these young students would need restrictions on what they can do and most of their actions would be directly guided by the teachers.  As students get older, teachers would likely provide more freedom.  In this case, a badging system might help control what people, rooms, and objects they can interact with.  The audio is excellent and very sophisticated when multiple groups are nearby.  This might be attractive to older K12 students, but it would be helpful to disable it for younger students.

**Comment by johnshaughnessy** [comment, marked as answer]
Hi @johnfaig-stpats ! Nice meeting you too. I've responded to some of your points below. Besides that, I would love to hear more details about how you would like to use Hubs in a classroom setting. Hubs-for-educational-use-cases has been a popular topic of conversation and exploration for the team lately, and we need more input from educators so we can dive deeper.
Thanks!

>  some of the power makes it difficult to use in a K12 environment.

This seems consistent with some of the ways people have been customizing the experience to meet their needs. It's common for people to _remove_ features rather than add more.

> Moving in the X, Y, and Z axes would be confusing for our younger students that don't play video games.

It is possible to lockdown movement using waypoints to restrict movement (`Disable movement` and `Disable teleporting`):
![image](https://user-images.githubusercontent.com/4072106/105742574-8c10b800-5ef0-11eb-95ee-fa9a1e1ac19f.png)

It is also possible (albiet inconvenient) to assign seats by removing the `Clickable` flag, naming each waypoint uniquely, and sending each student a unique link to the room with their assigned seat by specifying their seat with an "anchor hash". (See https://github.com/mozilla/hubs/pull/2837#issue-465665361 for details).

> a top-down view would look more like a classroom and still allow for different furniture configurations

We do not have a top-down view but I agree this could be very useful.

> young students would need restrictions on what they can do and most of their actions would be directly guided by the teachers.

> a badging system might help control what people, rooms, and objects they can interact with

There are some [room-wide permissions](https://hubs.mozilla.com/docs/hubs-room-settings.html#room-permissions) that can be set. We have begun to discuss expansions to the room / role permissions model. I thought we had a single meta-issue tracking this topic, but I can't seem to find it. Will follow up if I do find it. (You can see that this topic has been brought up many times if you search the discussions and issues.)

> The audio is excellent and very sophisticated when multiple groups are nearby. This might be attractive to older K12 students, but it would be helpful to disable it for younger students.

Can you expand on what you mean by this? Are you suggesting that the younger students might be in Hubs rooms without voice chat?

**Reply by johnshaughnessy** [reply]
By the way, if you haven't seen it yet -- I recommend you check out the `Hubs School v1.0` project in spoke that @cvanmeurs , an artist on the Hubs team, created for use cases like yours. Here's a screenshot:
![image](https://user-images.githubusercontent.com/4072106/105745397-10643a80-5ef3-11eb-8d50-d5f3ba0482b7.png)

You can find it in the featured project templates in Spoke.

**Reply by johnfaig-stpats** [reply]
With regard to the audio comment, I don't think the cocktail party effect is helpful in a school environment.

**Reply by johnfaig-stpats** [reply]
Wow!  The model built by @cvanmeurs looks intriguing.  I wonder if it also restricts some of the capabilities so that it is easy for a teacher to use.

**Reply by cvanmeurs** [reply]
> 
> 
> Wow! The model built by @cvanmeurs looks intriguing. I wonder if it also restricts some of the capabilities so that it is easy for a teacher to use.

Hey there.  I'm the creator of that scene.  Thanks for your interest in it.  :)  Any questions I can help you with?

**Reply by cvanmeurs** [reply]
You can check the scene out here:  https://hubs.mozilla.com/scenes/OojOIa3/hubs-school-v1-0

**Reply by johnfaig-stpats** [reply]
Do you think it would be easy to add capabilities and controls to a scene like your classroom?

**Reply by cvanmeurs** [reply]
What capabilities are you looking for?  I'm always open to suggestions.  This is the "1.0" version of the classroom/school scene, and I'd like to update it at some point to include some of the feedback I have received.  For example, one teacher in NY who has been using this for her class has requested that I add an office for her, which I will be doing in version 2.0.  We have a "virtual classrooms" channel on our discord server if you'd like to join the discussion there, although feel free to post here if that works better for you.

**Reply by johnfaig-stpats** [reply]
I checked out the Discord thread, but it is very specific.  My thoughts are more holistic and my critical issue is the question about Hubs being the right platform.

**Comment by robinkwilson** [comment]
I'm marking this answered for now.

If you have any more _**specific**_ questions about hubs and outline exactly what you're looking for, feel free to unmark the answer to put it back in the pipeline (or make a new discussion thread)!
